### PR TITLE
refactor: Overhaul 2D Godot docs

### PR DESCRIPTION
**Pull request by BoopEnthusiast**
Lots of changes here, a lot of restructuing. Especially adding a lot of things so that it's easy to find specific sections during lessons. That and just making it plain better

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "@astrojs/starlight-tailwind": "^2.0.1",
     "@astrojs/tailwind": "^5.1.0",
     "astro": "^4.0.8",
+    "pnpm": "^8.15.4",
```
Comment (Darkflame72) [inline package.json:17]: Why is this here?

**Review by Darkflame72 — CHANGES_REQUESTED** [review]
Got some great changes

File: package.json (hunk 1 above)
Comment (BoopEnthusiast) [reply package.json:17]: Not sure, I'll just copy paste the current package.json file into this branch's

Comment (Darkflame72) [reply package.json]: There shouldn't be a package-lock.json and the pnpm lock should not have been updated in this PR so can you revert those please so that those 3 files have no changes in this PR

Comment (BoopEnthusiast) [reply package.json]: I think I got it this time

**Review by Darkflame72 — APPROVED** [review]
